Source/WebCore/Modules/model-element/DDModelPlayer.mm

```diff
@@ -176,6 +214,10 @@ void setOpaque(bool opaque)
                     protectedThis->m_currentModel->update(toCpp(updateRequest));
 
                 [protectedThis->m_modelLoader requestCompleted:updateRequest];
+                if (RefPtr client = protectedThis->m_client.get()) {
+                    auto [center, extents] = computeCenterAndExtents(updateRequest.parts, updateRequest.instanceTransforms);
+                    client->didUpdateBoundingBox(protectedThis.get(), center, extents);
+                }
```

Comment:
This might be more correct. But I want to make sure we're mindful about breaking the parallel between this and the ModelProcess implementation (and potentially update the ModelProcess one when it makes sense). Especially if down the line this can help us avoid a third computeSRT implementation :) 

In ModelProcess land:
* The RKEntity computes the center and extents
* The Player(Proxy) keeps track of the original center+extents (to compute transforms later)
* `client->didUpdateBoundingBox(...)` only gets called on load right after `didFinishLoading()`

Could we do the computation in the ModelAdded callback based on the `addRequest`, and maybe move the code to DDMesh?

Reply:
I think `didFinishLoading` in DDModelPlayer.mm:
```
                if (RefPtr client = protectedThis->m_client.get())
                    client->didFinishLoading(protectedThis.get());
```
is called too early.

At the point `addMesh` returns, we aren't aware of the model's geometry as addMesh only contains the layout of the geometry and not the geometry itself.

Would the concern be addressed if we moved `didFinishLoading` to the updateMesh path, only calling it upon the first update?

Reply:
@etiennesegonzac I adopted the visionOS style and only send `didUpdateBoundingBox` right after `didFinishLoading`

Also moved the computations to the DDMeshProxy